### PR TITLE
Firebase integration2

### PR DESCRIPTION
**Pull request by nsathiya**
Able to pull file without call for uri (offline support added)

### CODE REVIEW

**Comment by nsathiya** [comment]
@rubengt01

File: src/index.tsx [hunk 1]
```diff
@@ -23,15 +25,17 @@ type CacheEntry = {
...
-        const ext = path.indexOf(".") === -1 ? ".jpg" : path.substring(path.indexOf("."));
-        if (immutable === true) {
-            return BASE_DIR + "/" + SHA1(uri) + ext;
-        } else {
-            return BASE_DIR + "/" + s4() + s4() + "-" + s4() + "-" + s4() + "-" + s4() + "-" + s4() + s4() + s4() + ext;
-        }
+    private getPath(dbPath: string, immutable?: boolean): string {
+      let path = dbPath.substring(dbPath.lastIndexOf("/"));
+      const terms = dbPath.split("/");
+      const toEncode = terms[terms.length-2];
```
Comment (rubengt01) [inline src/index.tsx:31]: /room1/logo/image.jpg
/room2/logo/image.jpg

both are going to have "logo" has value toEncode and the same name for SHA1(toEncode)

Comment (nsathiya) [reply src/index.tsx:31]: room1/[dbkey]/image.jpg. 
Hence dbkey is unique, and potentially I dont need to encode this either.

Comment (rubengt01) [reply src/index.tsx:31]: We are going to use this cache for all the images. It is true for the current images from chat messages, but we don't know if it is going to be true for future images and different implementations. Like company logo, User profiles, ...

**Pull request merged**